src/axios-har-tracker.ts

```diff
@@ -1,112 +1,168 @@
-          ),
-          httpVersion: `HTTP/${resp.request.res.httpVersion}`,
-          cookies: getCookies(JSON.stringify(resp.config.headers['Cookie'])),
-          bodySize: JSON.stringify(resp.data).length,
-          redirectURL: '',
-          headersSize: -1,
-          content: {
-            size: JSON.stringify(resp.data).length,
-            mimeType: resp.headers['content-type'] ? resp.headers['content-type'] : 'text/plain',
-            text: JSON.stringify(resp.data)
-          },
-          cache: {},
-          timings: {
-            blocked: -1,
-            dns: -1,
-            ssl: -1,
-            connect: -1,
-            send: 10,
-            wait: 10,
-            receive: 10,
-            _blocked_queueing: -1
-          }
-        };
-        const enteriesContent = Object.assign({}, newEntry);
-        generatedHar.log.entries.push(enteriesContent);
+export class AxiosHarTracker {
+
+  private axios: AxiosStatic;
+  private generatedHar: HarFile;
+  private date: string;
+  private newEntry: NewEntry;
+
+  constructor(axiosModule: AxiosStatic) {
+    this.axios = axiosModule;
+    this.generatedHar = {
+      log: {
+        version: '1.2',
+        creator: {
+          name: 'axios-har-tracker',
+          version: '0.1.0'
+        },
+        pages: [],
+        entries: []
+      }
+    };
+    this.date = new Date().toISOString();
+
+    this.axios.interceptors.request.use(
+
+      async config => {
+        this.newEntry = this.generateNewEntry();
+        this.newEntry.request = this.returnRequestObject(config);
+        return config;
+      },
+      async error => {
+        this.newEntry.request = this.returnRequestObject(error.request);
+        this.generatedHar.log.entries.push(this.newEntry);
+        return Promise.reject(error);
+      }
+    );
+
+    this.axios.interceptors.response.use(
+      async resp => {
+        this.pushNewEntryResponse(resp);
         return resp;
+      },
+      async error => {
+        this.pushNewEntryResponse(error.response);
+        return Promise.reject(error);
+      }
+    );
+  }
+
+  private returnRequestObject(config) {
+    const form = new URLSearchParams();
+    form.append('request-startTime', this.date);
+    form.append('request-duration', this.date);
+    const requestObject = {
+      method: config.method,
+      url: config.url,
+      httpVersion: 'HTTP/1.1',
+      cookies: this.getCookies(JSON.stringify(config.headers['Cookie'])),
+      headers: this.getHeaders(config.headers['common']),
+      queryString: this.getParams(config.params),
+      headersSize: -1,
+      bodySize: -1
+    };
+    return requestObject;
+  }
+
+  private returnResponseObject(response) {
+    const responseObject = {
+      status: response ? response.status : '',
+      statusText: response ? response.statusText : '',
+      headers: response ? this.getHeaders(response.headers) : {},
+      startedDateTime: this.date,
```

Comment:
Here you are assigning `this.date` which is generated once in constructor, this means that each response will have the same `startedDateTime`. You should create new date each time a response is received